pilot/pkg/kube/inject/app_probe.go

```diff
@@ -107,14 +121,22 @@ func rewriteAppHTTPProbe(spec *SidecarInjectionSpec, podSpec *corev1.PodSpec) {
 			port, exists := portMap[httpGet.Port.StrVal]
 			if !exists {
 				log.Errorf("named port not found in the map skip rewriting probing %v", *probe)
-				return
+				return nil
 			}
 			header.Value = strconv.Itoa(int(port))
 		}
 		httpGet.HTTPHeaders = append(httpGet.HTTPHeaders, header)
 		httpGet.Port = intstr.FromInt(statusPort)
+		if modify {
```

Comment:
IIUC `modify` is just `probe != nil`. Should we just use that condition here instead of the `modify` flag?

Reply:
`modify` can be true of false. 

- True for istioctl kube-inject case, where we just modify proto struct in-place.
- False for webhook case where we calculate the json patch instead.

Reply:
For the kube-inject case, it doesn't used the returned patch operation at all, right?

I think it's a bit confusing trying to combine these two code paths. It might be better to just keep them as separate code paths for now.  IIRC we eventually want to eventually have a single path for this (webhook), so a little duplication for now isn't a big deal.

Reply:
Yes, kube-inject case we don't need patch.

Done, although I'm not sure when the webhook will the single path for this.